src/Microsoft.Framework.CodeGeneration.Templating/Compilation/RoslynCompilationService.cs

```diff
@@ -59,82 +36,5 @@ public CompilationResult Compile(string content)
                 return CompilationResult.Failed(content, result.ErrorMessages);
             }
         }
-
-        //Todo: This is using application references to compile the template,
-        //perhaps that's not right, we should use the dependencies of the caller
-        //who calls the templating API.
-        private List<MetadataReference> GetApplicationReferences()
-        {
-            var references = new List<MetadataReference>();
-
-            // Todo: When the target app references scaffolders as nuget packages rather than project references,
-            // since the target app does not actully use code from scaffolding framework, our dlls
-            // are not in app dependencies closure. However we need them for compiling the
-            // generated razor template. So we are just using the known references for now to make things work.
-            // Note that this model breaks for custom scaffolders because they may be using
-            // some other references which are not available in any of these closures.
-            // As the above comment, the right thing to do here is to use the dependency closure of
-            // the assembly which has the template.
-            var baseProjects = new string[] { _environment.ApplicationName, "Microsoft.Framework.CodeGeneration" };
```

Comment:
Can you show me a template which would break? As we mentioned in our emails the ones in Mvc use a different mechanism and this does not break any of the existing templates.

The file that is produced just looks like this:

```public class __CompiledTemplate : RazorTemplateBase

    public void ExecuteAsync()
    {
        Write(xxxx);
        WriteLiteral(yyyy);
        ...
    }
```

The only thing that is referenced anywhere is Model which is a dynamic object defined on RazorTemplateBase.

As I say I found nothing that would depend on the project in which the template exists.